File: festival-web/src/main/java/cn/bdqfork/web/server/WebVerticle.java
package cn.bdqfork.web.server;

import io.vertx.core.AbstractVerticle;
import io.vertx.core.Promise;

/**
 * @author bdq
 * @since 2020/1/21
 */
public class WebVerticle extends AbstractVerticle {
    private WebServer webServer;

    public WebVerticle(WebServer webServer) {
        this.webServer = webServer;
    }

    @Override
    public void start(Promise<Void> startPromise) throws Exception {
        webServer.start();
        startPromise.complete();
    }

    @Override
    public void stop(Promise<Void> stopPromise) throws Exception {
        webServer.stop();
        stopPromise.complete();
    }

}

File: festival-kotlin-web/src/main/kotlin/cn/bdqfork/kotlin/web/server/WebVerticle.kt
package cn.bdqfork.kotlin.web.server

import io.vertx.kotlin.coroutines.CoroutineVerticle

/**
 * @author bdq
 * @since 2020/1/21
 */
class WebVerticle(private val webServer: WebServer) : CoroutineVerticle() {

    override suspend fun start() {
        webServer.start()
    }

    override suspend fun stop() {
        webServer.stop()
    }

}